packages/manager/src/components/PreferenceToggle/PreferenceToggle.tsx

```
@@ -18,171 +12,43 @@ interface RenderChildrenProps<T> {
-        } else if (
-          !!preferences &&
-          !isNullOrUndefined(currentlySetPreference) &&
-          lastUpdated !== 0
-        ) {
-          /**
-           * PUT to /preferences on every toggle, debounced.
-           */
-          updateUserPreferences({
-            [preferenceKey]: currentlySetPreference,
-          }).catch(() => /** swallow the error */ null);
-
-          /** invoke our callback prop if we have one */
-          if (
-            toggleCallbackFnDebounced &&
-            !isNullOrUndefined(currentlySetPreference)
-          ) {
-            toggleCallbackFnDebounced(currentlySetPreference);
-          }
-        } else if (lastUpdated === 0) {
-          /**
-           * this is the case where the app has just been mounted and the preferences are
-           * being set in local state for the first time
-           */
-          setLastUpdated(Date.now());
-        }
-      }, 500);
-
-      return () => clearTimeout(debouncedErrorUpdate);
-    }
-
-    return () => null;
-  }, [currentlySetPreference]);
-
   const togglePreference = () => {
-    /** first set local state to the opposite option */
-    const newPreferenceToSet =
-      currentlySetPreference === preferenceOptions[0]
-        ? preferenceOptions[1]
-        : preferenceOptions[0];
+    let newPreferenceToSet: T;
+
+    if (preferences?.[preferenceKey] === undefined) {
+      // Because we default to preferenceOptions[0], toggling with no preference should pick preferenceOptions[1]
+      newPreferenceToSet = preferenceOptions[1];
+    } else if (preferences[preferenceKey] === preferenceOptions[0]) {
+      newPreferenceToSet = preferenceOptions[1];
+    } else {
+      newPreferenceToSet = preferenceOptions[0];
+    }
 
-    /** set the preference in local state */
-    setPreference(newPreferenceToSet);
+    updateUserPreferences({
+      [preferenceKey]: newPreferenceToSet,
+    }).catch(() => /** swallow the error */ null);
```

Swallow the error? What does it mean for the end user?

---

It means their updated preferences won't be sent to the API. (this is existing code)

---

Right, and they won't know anything about it :( How about a TODO follow up item and associated ticket to display a toast (or whatever we decide) to handle this better?

---

Yeah, we could toast. I think we decided to swallow the errors because the idea of preferences is that is should be a low-impact failure if that makes any sense.

---

Created M3-7326